qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/template/TemplateFileTextDocumentService.java

```diff
@@ -204,32 +205,40 @@ public CompletableFuture<List<DocumentLink>> documentLink(DocumentLinkParams par
 
 	@Override
 	public CompletableFuture<List<Either<SymbolInformation, DocumentSymbol>>> documentSymbol(
-			DocumentSymbolParams params) {
+		DocumentSymbolParams params) {
 		return getTemplate(params.getTextDocument(), (cancelChecker, template) -> {
 			if (super.isHierarchicalDocumentSymbolSupport()) {
 				return getQuteLanguageService().findDocumentSymbols(template, cancelChecker) //
-						.stream() //
-						.map(s -> {
-							Either<SymbolInformation, DocumentSymbol> e = Either.forRight(s);
-							return e;
-						}) //
-						.collect(Collectors.toList());
-			}
-			return getQuteLanguageService().findSymbolInformations(template, cancelChecker) //
 					.stream() //
 					.map(s -> {
-						Either<SymbolInformation, DocumentSymbol> e = Either.forLeft(s);
+						Either<SymbolInformation, DocumentSymbol> e = Either.forRight(s);
 						return e;
 					}) //
 					.collect(Collectors.toList());
+			}
+			return getQuteLanguageService().findSymbolInformations(template, cancelChecker) //
+				.stream() //
+				.map(s -> {
+					Either<SymbolInformation, DocumentSymbol> e = Either.forLeft(s);
+					return e;
+				}) //
+				.collect(Collectors.toList());
 		});
 	}
 
 	@Override
 	public CompletableFuture<List<? extends Location>> references(ReferenceParams params) {
 		return getTemplate(params.getTextDocument(), (cancelChecker, template) -> {
 			return getQuteLanguageService().findReferences(template, params.getPosition(), params.getContext(),
-					cancelChecker);
+				cancelChecker);
+		});
+	}
+
+	@Override
+	public CompletableFuture<WorkspaceEdit> rename(RenameParams params) {

```

Comment:
In other words, you should just add this piece of code (other code should not be updated just for formatting).